pkg/cmd/projectCreate.go

```diff
@@ -0,0 +1,336 @@
+		TestArtifactId:   packageName + ".test",
+		OBRArtifactId:    packageName + ".obr",
+		Name:             packageName + ".obr",
+		GalasaVersion:    GALASA_VERSION}
+
+	targetFile := GeneratedFile{
+		fileType:                 "pom",
+		targetFilePath:           targetOBRFolderPath + "/pom.xml",
+		embeddedTemplateFilePath: "templates/projectCreate/parent-project/obr-project/pom.xml",
+		templateParameters:       pomTemplateParameters}
+
+	err := createFile(fileSystem, targetFile, forceOverwrite)
+	return err
+}
+
+// createFile creates a file on the file system.
+// If forceOverwrite is false, and there is already a file there, then an error will occur.
+func createFile(
+	fileSystem utils.FileSystem,
+	generatedFile GeneratedFile,
+	forceOverwrite bool) error {
+
+	log.Printf("Creating file of type %s at %s\n", generatedFile.fileType, generatedFile.targetFilePath)
+
+	err := checkAllowedToWrite(fileSystem, generatedFile.targetFilePath, forceOverwrite)
+	if err == nil {
+		var template *template.Template
+		template, err = loadEmbeddedTemplate(generatedFile.embeddedTemplateFilePath)
+		if err == nil {
+			var fileContents string
+			fileContents, err = substituteParametersIntoTemplate(template, generatedFile.templateParameters)
+			if err == nil {
+				// Write it out to the target file.
+				err = fileSystem.WriteTextFile(generatedFile.targetFilePath, fileContents)
+			}
+		}
+	}
+	if err == nil {
+		log.Printf("Created file %s OK.", generatedFile.targetFilePath)
+	}
+	return err
+}
+
+func substituteParametersIntoTemplate(template *template.Template, templateParameters interface{}) (string, error) {
+	// Render the golang template into a string
+	var buffer bytes.Buffer
+	fileContents := ""
+	err := template.Execute(&buffer, templateParameters)
+	if err == nil {
+		fileContents = buffer.String()
+	}
+	return fileContents, err
+}
+
+func loadEmbeddedTemplate(embeddedTemplateFilePath string) (*template.Template, error) {
+	// Load-up the template file from the embedded file system.
+	data, err := embeddedFileSystem.ReadFile(embeddedTemplateFilePath)
+	var templ *template.Template = nil
+	if err == nil {
+		// Parse the string data into a golang template
+		rawTemplate := template.New(embeddedTemplateFilePath)
+		templ, err = rawTemplate.Parse(string(data))
+	}
+	return templ, err
+}
+
+// checkAllowedToWrite - Checks to see if we are allowed to write a file.
+// The file may exist already. If it does, then we won't be able to over-write it unless
+// the forceOverWrite flag is true.
+func checkAllowedToWrite(fileSystem utils.FileSystem, targetFilePath string, forceOverwrite bool) error {
+	isAlreadyExists, err := fileSystem.Exists(targetFilePath)
+	if err == nil {
+		if isAlreadyExists && (!forceOverwrite) {
+			log.Printf("File %s exists, and we cannot over-write it as the --force flag is not set.", targetFilePath)
+			err = galasaErrors.NewGalasaError(galasaErrors.GALASA_ERROR_CANNOT_OVERWRITE_FILE, targetFilePath)
```

Comment:
Very picky error - BUT in the unlikely event that the file we are overwriting we no longer have permissions to write to then we fail:
```
Error: open com.hobbit.example/com.hobbit.example.test/src/main/java/com/hobbit/example/test/SampleTest.java: permission denied
Usage:
  galasactl project create [flags]

Flags:
      --force            Force-overwrite files which already exist.
  -h, --help             help for create
      --obr              An OSGi Object Bundle Resource (OBR) project is needed.
      --package string   Java package name for tests we create. Forms part of the project name, maven/gradle group/artifact ID, and OSGi bundle name. For example: com.myco.myproduct.myapp

Global Flags:
  -l, --log string   File to which log information will be sent. Any folder referred to must exist. An existing file will be over-written. Specify "-" to log to stderr. Defaults to not logging.

open com.hobbit.example/com.hobbit.example.test/src/main/java/com/hobbit/example/test/SampleTest.java: permission denied
```
it would be nice to wrapper that in a proper error message

Reply:
Wrapped such errors. They now look like this:
```
GAL1042E: Failed to open file 'com.myco.example/pom.xml' for writing. Reason is 'open com.myco.example/pom.xml: permission denied'. Check that you have permissions to write to that folder and file, and that there is enough disk space available and try again.
```

Reply:
It also doesn't echo out the usage now too.